File: ClientApp/src/app/employee/employee.ts
import { TimeSheet } from "../time-sheet/time-sheet";

export class Employee {
  constructor() {
    this.id = '';
    this.firstName = '';
    this.lastName = '';
    this.title = '';
    this.rate = 0;
    this.fedTax = 0;
    this.provTax = 0;
    this.rrq = 0;
    this.rqap = 0;
    this.vacation = 0;
    this.timeSheets = [];
  }
  public id: string = '';
  public firstName: string = '';
  public lastName: string = '';
  public title: string = '';
  public rate: number = 0;
  public fedTax: number = 0;
  public provTax: number = 0;
  public rrq: number = 0;
  public rqap: number = 0;
  public vacation: number = 0;

  public timeSheets?: TimeSheet[];

  public fullName(): string {
    return this.firstName + ' ' + this.lastName;
  }
  public isEditing: boolean = true;

}

File: Models/Employee.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SPA.Models
{
    public class Employee 
    {
        [Key]
        public string Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public ICollection<TimeSheet> TimeSheets { get; set; }

        /**HR**/
        public string Title { get; set; }
        public double Rate { get; set; }
        public double FedTax { get; set; }
        public double ProvTax { get; set; }
        public double RRQ { get; set; }
        public double RQAP { get; set; }
        public double Vacation { get; set; }

        [NotMapped]
        public string FullName => FirstName + " " + LastName;

    }
}
